### DISCUSSION TITLE
Question on link[rel=preload] for JS loaded via import()

### DISCUSSION THREAD

**Discussion by JamieMason** [vercel/next.js#15868] [Help] [answered]
I've been trying to reduce the bundle size of an App by Lazy-Loading Components using `'next/dynamic'`, then having those Components only appear in render tree when a particular button has been pressed for example.

I was curious to understand why a `<link rel="preload" href="/_next/static/chunks/hash.js" as="script"/>` is added for this category of Script, hinting to the browser to give them a high priority among all resources being downloaded.

My instinct is that this maybe shouldn't happen, because these Scripts would compete for resources alongside CSS, Images, Fonts etc. but I know there will be something I'm not aware of. I was just curious to learn more about what goes on here.

Thanks a lot.

**Comment by timneutkens** [comment, marked as answer]
The preload for those script tags is only included if you did in fact render the component using `next/dynamic`, so in the case that you're describing where it is only rendered after a button is clicked it would not preload.

**Reply by JamieMason** [reply]
Thanks a lot Tim, given what you've said I think I know what this is now in our codebase 👍
